torch/_inductor/lowering.py

```diff
@@ -4384,6 +4383,104 @@ def adaptive_max_pool2d(x, output_size):
     return rv, ri
 
 
+fallback_fractional_max_pool2d = fallback_handler(
+    aten.fractional_max_pool2d.default, add_to_fallback_set=False
+)
+
+
+def _fractional_pooling_offsets(samples, in_sz, out_sz, kernel_sz, dim):
+    out_sz = out_sz[dim]
+    in_sz = in_sz[dim]
+    kernel_sz = kernel_sz[dim]
+    alpha = (in_sz - kernel_sz) / (out_sz - 1)
+    samples_loader = samples.make_loader()
+
+    def load(prefix, i):
+        sample = ops.indirect_indexing(
+            samples_loader([*prefix, dim]), size=3, check=False
+        )
+
+        seq_i = ops.index_expr(
+            sympy.floor((i + sample) * alpha) - sympy.floor(sample * alpha), torch.int64
+        )
```

Comment:
This works, but it's a bit weird to do `indirect_indexing` followed by `index_expr`. We can just do everythign directly in the inductor IR and avoid `indirect_indexing`:
```suggestion
        sample = samples_loader([*prefix, dim])
        i_expr = ops.index_expr(i, samples.get_dtype())
        alpha_expr = ops.index_expr(alpha, samples.get_dtype())
        seq_i = ops.floor((i_expr + sample) * alpha) - ops.floor(sample * alpha)
        seq_i = ops.to_dtype(seq_i, torch.int64)
```